File: models/social/CommentModel.js
const mongoose = require('mongoose');

const CommentSchema = new mongoose.Schema({
    idcustomerId : {type:String,require:false},
    idPostId : {type:String,require:false},
    customerPhoto : {type:String,require:false},
    customerName : {type:String,require:false},
    title: { type: String, required: false,},
    dateNow:{type:String,default:Date.now()},
    areaPost:{type:String, require:false},
    content:{type:String, require:false},
    access:{type:Boolean, require:true},
    visible:{type:Boolean, require:true},
    position:{type:Number, require:false,default:1},
    coverPicture: { type: String, required: false}
}, 
{ timestamps: true }
);

const CommentModel = mongoose.model('comment_blog', CommentSchema);

module.exports = CommentModel;

File: lib/src/model/CommentModel.dart



class CommentModel {
  String? idcustomerId;
  String? idPostId;
  String? customerPhoto;
  String? customerName;
  String? title;
  String? dateNow;
  String? areaPost;
  String? content;
  bool? access;
  bool? visible;
  int? position;
  String? coverPicture;

  CommentModel({
    this.idcustomerId,
    this.idPostId,
    this.customerPhoto,
    this.customerName,
    this.title,
    this.dateNow,
    this.areaPost,
    this.content,
    this.access,
    this.visible,
    this.position,
    this.coverPicture,
  });

  factory CommentModel.fromJson(Map<String, dynamic> json) {
    return CommentModel(
      idcustomerId: json['idcustomerId'],
      idPostId: json['idPostId'],
      customerPhoto: json['customerPhoto'],
      customerName: json['customerName'],
      title: json['title'],
      dateNow: json['dateNow'],
      areaPost: json['areaPost'],
      content: json['content'],
      access: json['access'],
      visible: json['visible'],
      position: json['position'],
      coverPicture: json['coverPicture'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idcustomerId': idcustomerId,
      'idPostId': idPostId,
      'customerPhoto': customerPhoto,
      'customerName': customerName,
      'title': title,
      'dateNow': dateNow,
      'areaPost': areaPost,
      'content': content,
      'access': access,
      'visible': visible,
      'position': position,
      'coverPicture': coverPicture,
    };
  }
}
